spark/build.gradle.kts

```
@@ -53,6 +60,18 @@ android {
     kotlinOptions {
         jvmTarget = "11"
         allWarningsAsErrors = true
+        freeCompilerArgs = freeCompilerArgs + "-opt-in=com.adevinta.spark.InternalSparkApi"
+        freeCompilerArgs = freeCompilerArgs + "-opt-in=com.adevinta.spark.ExperimentalSparkApi"
+        freeCompilerArgs = freeCompilerArgs + "-opt-in=kotlin.RequiresOptIn"
```

IMO we should do that in a single statement to avoid creating unnecessary copies

```suggestion
        freeCompilerArgs += listOf(
            "-opt-in=com.adevinta.spark.InternalSparkApi",
            "-opt-in=com.adevinta.spark.ExperimentalSparkApi",
            "-opt-in=kotlin.RequiresOptIn",
        )
```

---

Agreed